### PR TITLE
DEV: add real time plottting for cdi code

### PR DESCRIPTION
**Pull request by licode**
Plotter module will be a general collector of easy plotting tools.

### CODE REVIEW

File: xray_vision/plotter.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    """
+
+    @classmethod
+    def from_axes(cls, ax0, ax1, ax2, ax3):
+        """
+        Set the axes from outside.
+
+        Parameters
+        ----------
+        ax0 : figure.axes
```
Comment (ericdill) [inline xray_vision/plotter.py]: `figure.axes` should be `matplotlib.axes.Axes`

File: xray_vision/plotter.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            from matplotlib
+        ax3 : figure.axes
+            from matplotlib
+        """
+        plot = CDIPlotter()
+        # clear the axes
+        ax0.cla()
+        ax1.cla()
+        ax2.cla()
+        ax3.cla()
```
Comment (ericdill) [inline xray_vision/plotter.py]: @tacaswell is this kosher?

Comment (ericdill) [inline xray_vision/plotter.py]: by "this" I mean, is it generally ok to clear the axes that a user passes in?

Comment (tacaswell) [inline xray_vision/plotter.py]: It's a little dicey, it means users can not use this to overlay multiple calls

Comment (ericdill) [inline xray_vision/plotter.py]: This was added at your suggestion, would you prefer it to be taken out?

Comment (ericdill) [inline xray_vision/plotter.py]: Errr, this [class] was added at your suggestion. Would you prefer [these cla's] to be taken out?

Comment (tacaswell) [inline xray_vision/plotter.py]: Just document that it will happen.

**Comment by licode** [comment]
Any more comments?

**Comment by licode** [comment]
Ready to go? @tacaswell

**Comment by tacaswell** [comment]
Typically the way this pattern works is

```python
class foo:

    @classmethod
    def magic_setup(cls, some, args):
        cranky, details, things = helpful_function(some, args)
        return foo(cranky, details, things)

    def __init__(self, cranky, details, things):
        self.cranky = cranky
        self.details = details
        self.things = things
        self.other_thing = self.cranky.make_artist()

    def work_function(self, data):
        self.other_thing.set_data(data)
```

**Comment by ericdill** [comment]
@licode what is the status of this?

**Comment by ericdill** [comment]
@licode were there any other changes you wanted to make to this?

**Comment by licode** [comment]
good to me.

**Pull request merged**